WPFHospitalEditor/HospitalMap.xaml.cs

```diff
@@ -135,7 +140,34 @@ public void Medication_Search(object sender, RoutedEventArgs e)
 
         public void appointmentSearch_Click(object sender, RoutedEventArgs e)
         {
+            clearAllResults();
+            if (InvalidInputForAppointment())
+            {
+                MessageBox.Show("Invalid input.");
+            }
+            else
+            {
+                int index = doctorsComboBox.SelectedIndex - 1;
+                int doctorId = doctorServerController.GetDoctorsByDepartment(regularExaminationDepartment).ElementAt(index).DoctorId;
+
+                RecommendationRequestDto recommendationRequestDto = new RecommendationRequestDto()
+                {
+                    DoctorId = doctorServerController.GetDoctorsByDepartment(regularExaminationDepartment).ElementAt(index).DoctorId,
```

Comment:
I'm not sure that this part of code is working properly. Searching for results via a selected index can produces unplanned errors. For this to work you have to make sure that order within the combobox matches the order of the function return.